drivers/adc/adc_mcux_lpadc.c

```diff
@@ -571,23 +545,14 @@ static const struct adc_driver_api mcux_lpadc_driver_api = {
 			  "Invalid clock source");			\
 	ASSERT_LPADC_CLK_DIV_VALID(DT_INST_PROP(n, clk_divider),	\
 		   "Invalid clock divider");			\
-	ASSERT_WITHIN_RANGE(DT_INST_PROP(n, voltage_ref), 2, 3,	\
-		"Invalid voltage reference source");	\
-	ASSERT_LPADC_CALIBRATION_AVERAGE_VALID(		\
-		DT_INST_PROP(n, calibration_average),	\
-		"Invalid conversion average number for auto-calibration time");	\
-	ASSERT_WITHIN_RANGE(DT_INST_PROP(n, power_level), 1, 4,		\
-		"Invalid power level");					\
 	PINCTRL_DEFINE(n)						\
 	static const struct mcux_lpadc_config mcux_lpadc_config_##n = {	\
 		.base = (ADC_Type *)DT_INST_REG_ADDR(n),	\
 		.clock_source = TO_LPADC_CLOCK_SOURCE(DT_INST_PROP(n, clk_source)),	\
 		.clock_div = DT_INST_PROP(n, clk_divider),					\
-		.voltage_ref =												\
-			TO_LPADC_REFERENCE_VOLTAGE(DT_INST_PROP(n, voltage_ref)),	\
-		.calibration_average =										\
-			TO_LPADC_CALIBRATION_AVERAGE(DT_INST_PROP(n, calibration_average)),	\
-		.power_level = TO_LPADC_POWER_LEVEL(DT_INST_PROP(n, power_level)),	\
+		.voltage_ref =	DT_INST_PROP(n, voltage_ref),	\
```

Comment:
Let's add a comment block here that highlights that the voltage_ref and calibration_average is given in units as defined in the reference manual for future users.

Reply:
I think I will put something like this in the binding so that it will be visible in menuconfig